Source/WebCore/css/parser/CSSSelectorParser.cpp

```diff
@@ -43,29 +43,36 @@ namespace WebCore {
 static AtomString serializeANPlusB(const std::pair<int, int>&);
 static bool consumeANPlusB(CSSParserTokenRange&, std::pair<int, int>&);
 
-std::optional<CSSSelectorList> parseCSSSelector(CSSParserTokenRange range, const CSSParserContext& context, StyleSheetContents* styleSheet)
+std::optional<CSSSelectorList> parseCSSSelector(CSSParserTokenRange range, const CSSParserContext& context, StyleSheetContents* styleSheet, bool isNestedSelector)
 {
-    CSSSelectorParser parser(context, styleSheet);
+    CSSSelectorParser parser(context, styleSheet, isNestedSelector);
     range.consumeWhitespace();
     CSSSelectorList result = parser.consumeComplexSelectorList(range);
     if (result.isEmpty() || !range.atEnd())
         return { };
     return result;
 }
 
-CSSSelectorParser::CSSSelectorParser(const CSSParserContext& context, StyleSheetContents* styleSheet)
+CSSSelectorParser::CSSSelectorParser(const CSSParserContext& context, StyleSheetContents* styleSheet, bool isNestedSelector)
     : m_context(context)
     , m_styleSheet(styleSheet)
+    , m_isNestedSelector(isNestedSelector)
 {
 }
 
 CSSSelectorList CSSSelectorParser::consumeComplexSelectorList(CSSParserTokenRange& range)
 {
     Vector<std::unique_ptr<CSSParserSelector>> selectorList;
     auto selector = consumeComplexSelector(range);
-    if (!selector)
+    if (!selector) {
+        if (m_context.cssNestingEnabled) {
+            if (m_isNestedSelector && range.peek().type() == SemicolonToken) {
+                range.consumeIncludingWhitespace();
+                return { };
```

Comment:
Even if none of the above it true, we always fall through to the '`return { };`' below. Maybe this early return isn't that useful?